assets/src/components/app.tsx

```diff
@@ -15,30 +25,47 @@ import { OpenView } from "../state"
 const AppRoutes = () => {
   useAppcues()
 
-  const [{ pickerContainerIsVisible, openView }] =
+  const [{ pickerContainerIsVisible, openView, selectedRouteIds }] =
     useContext(StateDispatchContext)
 
+  const { socket }: { socket: Socket | undefined } = useContext(SocketContext)
+
+  const location = useLocation()
+
+  const vehiclesByRouteId: ByRouteId<VehicleOrGhost[]> = useVehicles(
+    socket,
+    openView === OpenView.Late || location.pathname === "/"
```

Comment:
@lemald it took me a moment to understand what's going on here, let's add an explaining variable:

```
const vehiclesByRouteIdNeeded = openView === OpenView.Late || location.pathname === "/"

const vehiclesByRouteId: ByRouteId<VehicleOrGhost[]> = useVehicles(
  socket,
  vehiclesByRouteIdNeeded ? selectedRouteIds : []
)
```

Reply:
Yeah good call.